File: lib-segment/src/main/java/websiteschema/mpsegment/graph/Path.java
package websiteschema.mpsegment.graph;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class Path
        implements Cloneable {

    public Path() {
        vertexList = new ArrayList();
    }

    public Path(List<Integer> list) {
        vertexList = list;
    }

    public void Clear() {
        vertexList.clear();
    }

    public int getLength() {
        return vertexList.size() - 1;
    }

    public Path addVertex(int vertex) {
        vertexList.add(vertex);
        return this;
    }

    public Path addPath(Path path) {
        vertexList.addAll(path.vertexList);
        return this;
    }

    public Object get(int i) {
        return vertexList.get(i);
    }

    public int iget(int i) {
        Integer integer = (Integer) vertexList.get(i);
        return integer.intValue();
    }

    public Object getLast() {
        return vertexList.get(vertexList.size() - 1);
    }

//    @Override
//    public Object clone() throws CloneNotSupportedException {
//        Path path = null;
//        try {
//            path = (Path) super.clone();
//        } catch (CloneNotSupportedException ex) {
//            throw ex;
//        }
//        path.vertexList = (ArrayList) vertexList.clone();
//        return path;
//    }

    @Override
    public String toString() {
        String s = new String();
        Iterator iterator = vertexList.iterator();
        if (iterator.hasNext()) {
            Object obj = iterator.next();
            s = (new StringBuilder()).append(s).append(obj.toString()).toString();
        }
        while (iterator.hasNext()) {
            Object obj1 = iterator.next();
            s = (new StringBuilder()).append(s).append("-").append(obj1.toString()).toString();
        }
        return s;
    }
    private List<Integer> vertexList;
}

File: lib-segment/src/main/scala/me/yingrui/segment/graph/Path.scala
package me.yingrui.segment.graph

import collection.mutable.ListBuffer

class Path {

  def Clear() {
    vertexList.clear()
  }

  def getLength(): Int = {
    return vertexList.size - 1
  }

  def addVertex(vertex: Int): Path = {
    vertexList += (vertex)
    return this
  }

  def addPath(path: Path): Path = {
    vertexList ++= (path.vertexList)
    return this
  }

  def get(i: Int): Int = {
    return vertexList(i)
  }

  def iget(i: Int): Int = {
    return vertexList(i)
  }

  def getLast(): Int = {
    return vertexList(vertexList.size - 1)
  }

  override def toString(): String = {
    val s = new StringBuilder()

    for (vertex <- vertexList) {
      s.append(vertex).append("-")
    }
    if (s.endsWith("-")) {
      s.deleteCharAt(s.length - 1)
    }
    return s.toString()
  }

  private var vertexList = ListBuffer[Int]()
}

object Path {
  def apply() = new Path()

  def apply(list: List[Int]) = {
    val p = new Path
    p.vertexList ++= list
    p
  }
}
